pwndbg/arguments.py

```diff
@@ -83,7 +83,17 @@ def get(instruction: PwndbgInstruction) -> List[Tuple[pwndbg.lib.functions.Argum
         name = name.replace("_chk", "")
         name = name.strip().lstrip("_")  # _malloc
 
-    func = pwndbg.lib.functions.functions.get(name, None)
+    # Try to resolve an Objective-C method call.
+    #
+    # Checking this first keeps us from resolving these as simple calls to
+    # `objc_msgSend` and functions like it, which have definitions that are
+    # rather barren of semantics in comparison.
+    func = pwndbg.aglib.objc.try_resolve_call_at_current_pc(instruction)
```

Comment:
I am not sure I like this.

Can we somehow detect obj-c language? Does it show with `show language`? Maybe GDB does auto-detect it? (Does LLDB do so? :/)

Reply:
<img width="362" height="142" alt="image" src="https://github.com/user-attachments/assets/05b21e84-bbde-4752-9fef-3c343928fe90" />

Reply:
Alternatively, we could do this code only when running on macOS/Darwin? Does Obj-C code ever run on Linux?:D

Reply:
> yes you can run obj-c on linux

Honestly I don't think that matters much? `pwndbg.aglib.objc` is currently only concerned with the Darwin version of the ABI, which isn't exactly binary compatible with GNUStep, afaik.

Reply:
we should add language checking before executing `pwndbg.aglib.objc.*` code